components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/SecondaryUserStoreConfigurationUtil.java

```diff
@@ -339,6 +339,7 @@ private static void updateUserStoreProperties(Path userStoreConfigFile, UserStor
         StreamResult result = new StreamResult(Files.newOutputStream(userStoreConfigFile));
         DOMSource source = new DOMSource(doc);
         transformProperties().transform(source, result);
+        result.getOutputStream().close();
```

Comment:
shall we do try catch and print a log.error for file close issue. and throw the excpetion?

Reply:
Changed the code as we discussed offline.